packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -206,12 +206,21 @@ export class InternalModelManager {
     return record ?? null;
   }
 
-  async findMany(options?: Record<string, any>, throwOnEmptyData = true, isFirstQuery = false): Promise<GadgetRecordList<any>> {
+  async findMany(options?: Record<string, any>, throwOnEmptyData?: boolean, isFirstQuery = false): Promise<GadgetRecordList<any>> {
     const response = await this.connection.currentClient
       .query(internalFindManyQuery(this.apiIdentifier, isFirstQuery), options)
       .toPromise();
-    const assertSuccess = throwOnEmptyData ? assertOperationSuccess : assertNullableOperationSuccess;
-    const connection = assertSuccess(response, ["internal", `list${this.capitalizedApiIdentifier}`]);
+
+    let connection;
+    if (throwOnEmptyData === false) {
+      // If this is a nullable operation, don't throw errors on empty
+      connection = assertNullableOperationSuccess(response, ["internal", `list${this.capitalizedApiIdentifier}`]);
+    } else {
+      // Otherwise, passthrough the `throwOnEmptyData` flag, to account for
+      // `findMany` (allows empty arrays) vs `findFirst` (no empty result) usage.
+      connection = assertOperationSuccess(response, ["internal", `list${this.capitalizedApiIdentifier}`], throwOnEmptyData);
```

Comment:
Would it? From what I saw in `assertOperationSuccess`, it would only throw id the data was `undefined` or `null`, which I'm assuming it would never be for a `findMany` operation.

In any case, I'm not picky about this at all, but though a search for a list would be more reasonably expected to return an empty array (rather than an error).

Either way, if you're okay with it like this, I'll go ahead and merge.